src/main/java/org/humancellatlas/ingest/state/StateMachineConfiguration.java

```diff
@@ -310,11 +310,15 @@ public void configure(StateMachineTransitionConfigurer<SubmissionState, Submissi
 
                 documentTracker = new DocumentTracker(documentCount);
                 context.getExtendedState().getVariables().put(tracker, documentTracker);
+            } else if(targetState.equals(MetadataDocumentState.PROCESSING)) {
+                int documentCount = context.getMessageHeaders().get(EXPECTED_DOCUMENT_COUNT, Integer.class);
+                documentTracker.reset(documentCount);
```

Comment:
This might not work if there's a processing message sent after a complete message.

e.g. 
assay 1 processing
assay 2 processing
assay 3 processing
assay 1 complete
assay 4 processing -- at this point the assay 1 complete message's increment in completedDocumentCount is disregarded because completedDocumentCount is reset to 0
assay 2 complete
assay 3 complete
assay 4 complete

I don't know if this scenario is unlikely. But, I think the tracker obj should be reset after the submission is set to Submitted/Exported not when it sees an _assay processing_ message.

Reply:
I modified the test for this scenario and applied some changes. Jacob agreed offline. :)